apps/demo/app/[...puckPath]/client.tsx

```diff
@@ -52,15 +47,14 @@ export function Client({ path, isEdit }: { path: string; isEdit: boolean }) {
         <Puck
           config={config}
           data={data}
-          onPublish={async (data: Data) => {
-            localStorage.setItem(key, JSON.stringify(data));
-          }}
+          onChange={(data) => setLocalData(data)}
+          onPublish={(data) => publishPageData(path, data)}
```

Comment:
I don't know if having a separate localStorage-powered "preview" route and separate published route makes sense here, because the publish route can never be updated by the user.

The point of the demo app is to act like a sandbox for a how real application might work, and the fake "publishing" workflow is important to that.

_Having said that, we have been thinking about an official Puck website. The thought crossed my mind that we could make the homepage the demo application, in which case a preview mode might make sense (but this isn't something I want to solve now)_

Reply:
see my comments below, but happy to revert to the local storage solution